### ISSUE TITLE
Android build: `java.lang.RuntimeException: Duplicate class`

### ISSUE THREAD

**Issue by joao-conde** [rodgomesc/vision-camera-code-scanner#49]
Hello,

I am using this package. I have added it to my package.json:

```json
{
    "name": "XXXXXX",
    "version": "XXXXX",
    "private": true,
    "scripts": {
        "android": "react-native run-android",
        "android-release": "react-native run-android --variant=release",
        "cleanup": "rm -rf node_modules && rm -f package-lock.json && rm yarn.lock",
        "clear": "react-native start --reset-cache",
    },
    "jest": {
        "preset": "react-native"
    },
    "dependencies": {
        "@react-native-async-storage/async-storage": "^1.16.1",
        "@react-native-community/clipboard": "^1.5.1",
        "@react-native-community/cookies": "^5.0.1",
        "@react-native-community/datetimepicker": "^5.1.0",
        "@react-native-community/masked-view": "^0.1.11",
        "@react-native-community/netinfo": "^8.0.0",
        "@react-native-community/picker": "^1.8.1",
        "@react-navigation/bottom-tabs": "^6.2.0",
        "@react-navigation/native": "^6.0.8",
        "@react-navigation/stack": "^6.1.1",
        "@sentry/react-native": "^3.2.13",
        "buffer": "^6.0.3",
        "fast-text-encoding": "^1.0.3",
        "grapheme-splitter": "^1.0.4",
        "js-base64": "^3.7.2",
        "pusher-js": "^7.0.6",
        "react": "^17.0.2",
        "react-native": "^0.67.2",
        "react-native-device-info": "^8.4.9",
        "react-native-document-picker": "^8.0.0",
        "react-native-fs": "^2.19.0",
        "react-native-gesture-handler": "^2.2.0",
        "react-native-image-picker": "4.7.3",
        "react-native-linear-gradient": "^2.5.6",
        "react-native-onesignal": "^4.3.7",
        "react-native-picker-select": "^8.0.4",
        "react-native-reanimated": "^2.4.1",
        "react-native-safe-area-context": "^3.3.2",
        "react-native-safe-area-view": "^1.1.1",
        "react-native-screens": "^3.12.0",
        "react-native-svg": "^12.1.1",
        "react-native-vision-camera": "^2.13.0",
        "react-native-webview": "^11.17.2",
        "ripe-commons": "^0.8.1",
        "ripe-commons-native": "^0.4.0",
        "ripe-components-react-native": "^0.19.0",
        "ripe-id-api": "^0.6.1",
        "ripe-sdk": "^2.20.0",
        "url": "^0.11.0",
        "vision-camera-qrcode-scanner": "^0.1.4",
        "yonius": "^0.11.5"
    }
}
```

And to my `android/app/build.gradle` I have tried:

```gradle
missingDimensionStrategy "react-native-vision-camera", "general"
```

and

```gradle
missingDimensionStrategy "vision-camera-qrcode-scanner", "general"
```

and

```gradle
missingDimensionStrategy "react-native-vision-camera", "mlkit"
```

and

```gradle
missingDimensionStrategy "vision-camera-qrcode-scanner", "mlkit"
```

None work, all fail when running `react-native run-android` with:

```
* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> 1 exception was raised by workers:
  java.lang.RuntimeException: Duplicate class com.google.firebase.components.AbstractComponentContainer found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component$$Lambda$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component$$Lambda$2 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component$$Lambda$3 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Component$Builder found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentContainer found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentDiscovery found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentDiscovery$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentDiscovery$MetadataRegistrarNameRetriever found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentDiscovery$RegistrarNameRetriever found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentFactory found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentRegistrar found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentRuntime found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentRuntime$$Lambda$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentRuntime$$Lambda$2 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.ComponentRuntime$$Lambda$3 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.CycleDetector found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.CycleDetector$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.CycleDetector$ComponentNode found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.CycleDetector$Dep found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Dependency found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.DependencyCycleException found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.DependencyException found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.EventBus found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.EventBus$$Lambda$1 found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.Lazy found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.MissingDependencyException found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.RestrictedComponentContainer found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.components.RestrictedComponentContainer$RestrictedPublisher found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.events.Event found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.events.EventHandler found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.events.Publisher found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.events.Subscriber found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  Duplicate class com.google.firebase.inject.Provider found in modules jetified-firebase-common-18.0.0-runtime.jar (com.google.firebase:firebase-common:18.0.0) and jetified-firebase-components-16.1.0-runtime.jar (com.google.firebase:firebase-components:16.1.0)
  
  Go to the documentation to learn how to <a href="d.android.com/r/tools/classpath-sync-errors">Fix dependency resolution errors</a>.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 14s

    at makeError (/home/jotac/Documents/ripe-robin-revamp/node_modules/execa/index.js:174:9)
    at /home/jotac/Documents/ripe-robin-revamp/node_modules/execa/index.js:278:16
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async runOnAllDevices (/home/jotac/Documents/ripe-robin-revamp/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:109:5)
    at async Command.handleAction (/home/jotac/Documents/ripe-robin-revamp/node_modules/@react-native-community/cli/build/index.js:192:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I tried adding the snippet recommended to `babel.config.js` but it also does not work. Also, I think I don't need this because I don't want to use right now the `'react-native-reanimated'` import.

**Comment by joao-conde** [comment]
I have tried clean installing everything and using the `react-native-reanimated` import and the `babel.config.js` change but nothing works. I still get the same error for duplicate class, this makes no sense. Anyone been able to reproduce this?

**Comment by rodgomesc** [comment]
hi @joao-conde  sorry for the delay, i'm facing this bug as well, i would like to suggest we work on the same example project in this repo, so we can be close as possible in our enviroment and and make sure there are no other external factors involved in the crash,  i'm digging on this  and for now the the only clue i have is that it seems to be involved with reanimated lib

i'll keep you updated

![Screen Shot 2022-04-15 at 10 55 59](https://user-images.githubusercontent.com/4893591/163580537-9f3e2c27-600b-4088-b3c1-5b7d52f9e1ab.png)

**Comment by rodgomesc** [comment]
hi @joao-conde  sorry for the delay, can you confirm that the example project in this repo build successfully?

**Comment by joao-conde** [comment]
Hi @rodgomesc, you can close this one

The problem was another dependency used a different firebase version which I guess ends up trying to build two firebase versions which gives a duplicate class error.

Adding 

```
implementation platform('com.google.firebase:firebase-bom:29.1.0') 
```

to `android/app/build.gradle` fixes a version and fixes the problem 👍

**Comment by congshengwu** [comment]
Hi @joao-conde, thanks for your workaround, it resolved my issue. From my side, I manually linked `react-native-vision-camera` and `vision-camera-code-scanner` into my project and got this error.

I also searched for this error `Duplicate class` on StackOverflow, and found a similar [solution](https://stackoverflow.com/questions/67546931/google-ml-kit-duplicate-dependency).

I integrated firebase in my project, and I think there is a different version of firebase in the transitive dependency `com.google.mlkit:barcode-scanning` of `
vision-camera-code-scanner`.
